packages/x-charts/src/context/CartesianContextProvider.tsx

```diff
@@ -236,6 +245,15 @@ function CartesianContextProvider(props: CartesianContextProviderProps) {
               ? getOrdinalColorScale({ values: axis.data, ...axis.colorMap })
               : getColorScale(axis.colorMap)),
         };
+        if (axis.data?.[0] instanceof Date) {
+          const timeScale = scaleTime(axis.data!, range);
+          completedXAxis[axis.id].valueFormatter =
+            axis.valueFormatter ??
+            ((v, { location }) =>
+              location === 'tick'
+                ? timeScale.tickFormat(axis.tickNumber)(v)
+                : `${v.toLocaleString()}`);
```

Comment:
We could probably also create a `dateValueFormatter` and just `completedXAxis[axis.id].valueFormatter = dateValueFormatter(...)`